docs/proposals/multiple-sources-for-applications-ui.md

```diff
@@ -0,0 +1,226 @@
+
+2. History and Rollback is disabled for multi-source applications. The button is disabled. Jorge has submitted a PR for 
+rollback which includes [controller and UI changes](https://github.com/argoproj/argo-cd/pull/14124).
+
+   <img height="40%" width="40%" src="images/history-and-rollback-button.png"/>
+
+3. The New Application dialog currently only allows users to provide one source.
+
+Thus, multiple source applications are not considered first class citizens in the UI.
+
+Note, see the [Open Questions](https://github.com/argoproj/argo-cd/docs/proposals/multiple-sources-for-applications-ui.md#open-questions) 
+section for concerns regarding the priority or value of some of the above changes.
+
+## Motivation
+
+The motivation behind this change is to add a more complete story for the multiple source feature. The UI should support 
+the creation of multiple source applications, and also support the viewing and editing of parameters from all sources. The three
+points in the summary above are the base or core changes that need to be addressed.
+
+### Goals
+
+The goals of the proposal are:
+
+- Provide first-class support of multiple sources for applications in the UI (e.g. address the aforementioned restrictions)
+- Outline stages of implementation that will help ease PR review, and reduce the risk of introducing regressions/issues.
+
+
+### Non-goals
+* The design changes for the Argo CD CLI is beyond the scope of this proposal (The server APIs can probably be reused)
+
+## Proposal
+
+As mentioned in the previous summary section, the application source parameters are surfaced in the UI in three locations.
+The Resource details pages, specifically, the Summary and Parameters tabs, the deployment history, and the Application
+Create panel page. These pages should be updated.
+
+### Resource Details
+
+The following describes the current behavior and proposed changes for the Summary tab and the Parameters Tab.
+
+#### i) Summary Tab
+
+_Current Behavior:_ 
+
+The current Summary tab includes source-related information, including the repository. For example, in Figure 1 below,
+the REPO URL and PATH.
+
+<img height="50%" width="50%" src="images/current-summary-tab.png"/>
+
+Figure 1: The current Summary tab
+
+_Proposed Change:_
+
+To support multiple sources, the source-related information, from a single-source-based design, will be ‘pulled out’ 
+and put into a new tab called **Sources**, and it will be combined with the **Parameters** tab (more details following). 
+The new **Sources** tab will allow users to view all the information related to each source, including the repo URL 
+and path, chart and revision for Helm, etc. 
+
+The view should show one source at a time (similar to what the UI is doing now, which only shows one source), but with 
+widgets to allow users to cycle (via pagination or combo selector?) through each source. There are API calls to retrieve 
+the data for each source.
```

Comment:
Just to clarify the proposal based on what was discussed above, does this make sense?

```suggestion
The view should show all the different sources in individual collapsible widgets with pagination to allow users to cycle through each source. 
There are API calls to retrieve the data for each source.

When there is only one source is the view, it will be expanded by default. If multiple sources are defined, they will be collapsed and the user will be able to expand them and paginate based on the number of items per page and page number. The default number of item per page should be 3 and saved in the user session.
```

Reply:
Regarding the comment "There are API calls to retrieve the data for each source":  I looked into this some more and I think for better performance and usability,  it should load the source details 'on demand' when the source section is expanded.  Like this:

Just shortly after clicking the expand button:
<img width="675" alt="LoadingOneSource" src="https://github.com/argoproj/argo-cd/assets/7726022/e46d3a59-e657-486e-8e92-f3055102000c">

Loading complete:
<img width="675" alt="LoadedOneSource" src="https://github.com/argoproj/argo-cd/assets/7726022/6570c815-2a80-40e6-a31b-b28042e527cc">

Reply:
LGTM, I though the default could have been configurable easily. I like the async loading, I was expecting the data to be already fetched, because it is part of the Application Manifest information, but if it needs to load something, async is definitely my preference.